File: friend_lover_nothing/lib/model/account.dart
import 'dart:core';
import 'package:json_annotation/json_annotation.dart';
part 'account.g.dart';

//model for API call- Account
@JsonSerializable()
class Account {
  String id;
  String firstName;
  String lastName;
  int? age;
  String? link;
  String? description;
  String email;
  String? phone;
  String? gender;
  String? profileImageLocation;
  String? interest1;
  String? interest2;
  String? interest3;

  Account(
      {required this.id,
      required this.firstName,
      required this.lastName,
      this.gender,
      this.age,
      this.link,
      this.description,
      required this.email,
      this.phone,
      this.profileImageLocation,
      this.interest1,
      this.interest2,
      this.interest3});

  factory Account.fromJson(Map<String, dynamic> json) =>
      _$AccountFromJson(json);

  Map<String, dynamic> toJson() => _$AccountToJson(this);

  Account.clone(Account account)
      : this(
            id: account.id,
            firstName: account.firstName,
            lastName: account.lastName,
            gender: account.gender,
            age: account.age,
            link: account.link,
            description: account.description,
            email: account.email,
            phone: account.phone,
            profileImageLocation: account.profileImageLocation,
            interest1: account.interest1,
            interest2: account.interest2,
            interest3: account.interest3);
}

File: FLN-Backend/src/main/java/com/neu/cs5520/flnbackend/model/Account.java
package com.neu.cs5520.flnbackend.model;

import java.io.Serializable;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Account implements Serializable {

  @Id
  @Column(length = 50, nullable = false)
  private String id;

  @Column(length = 62, nullable = false)
  private String email;

  @Column(length = 50)
  private String firstName;

  @Column(length = 50)
  private String lastName;

  int age;

  private String link;

  @Column(length = 100)
  private String description;

  @Column(length = 26)
  private String phone;

  @Column(length = 15)
  private String gender;

  private String profileImageLocation;

  @Column(length = 50)
  private String interest1;

  @Column(length = 50)
  private String interest2;

  @Column(length = 50)
  private String interest3;

}
